Assets/MixedRealityToolkit-SDK/Features/Input/FocusProvider.cs

```diff
@@ -801,6 +796,11 @@ private bool RaycastGraphicsStep(PointerData pointerData, RayStep step, LayerMas
             {
                 if (pointerData.CurrentPointerTarget != null)
                 {
+                    float distance = 0f;
+                    for (int i = 0; i <= pointer.RayStepIndex; i++) {
+                        distance += pointer.Pointer.Rays[i].Length;
```

Comment:
I think this will overshoot the true distance of the raycast, since the hit might be somewhere in the middle of the final ray step. For small raysteps, it'll probably be negligible, but it might cause a problem with larger ray steps. Not fully sure if that'll cause an issue with this calculation though.